Modules/Numerics/Optimizersv4/src/itkLBFGS2Optimizerv4.cxx

```diff
@@ -139,17 +141,29 @@ LBFGS2Optimizerv4::EvaluateCost(const LBFGS2Optimizerv4::PrecisionType * x,
                                 const LBFGS2Optimizerv4::PrecisionType)
 {
   ParametersType xItk(n);
-  std::memcpy(xItk.data_block(), x, n * sizeof(LBFGS2Optimizerv4::PrecisionType));
+  // TODO: check if this works with SSE allocation
+  // might need to copy data
+  xItk.SetData(const_cast<double *>(x), n, false);
```

Comment:
This is potentially dangerous but would avoid copying the data multiple times.
Is there are reason the SetParameters methods in the v4 metrics is not taking a const ParameterType?